conbench/tests/entities/test_commit.py

```diff
@@ -103,7 +107,15 @@ def test_get_github_commit_and_fork_point_sha(branch):
         # this is the master branch, so the fork point sha == the commit sha
         "fork_point_sha": sha,
     }
-    assert get_github_commit(repo, branch=branch, sha=sha, pr_number=None) == expected
+    try:
+        result = get_github_commit(repo, branch=branch, sha=sha, pr_number=None)
+    except Exception as exc:
+        log.info("exc during get_github_commit(): %s", exc)
+        if "API rate limit" in str(exc):
+            pytest.skip("GitHub API rate limit seen, skip test")
```

Comment:
Yep I really like this.

Reply:
This back-fired: https://github.com/conbench/conbench/issues/637#issue-1562612211

Did not remember this. Test suite passed because we skipped this test because of a rate limiting error. However, the patch was bad and in absence of rate limiting errors the test started "working again" in `main` (after merge), and then failed, exposing the bug.